packages/relay/src/lib/relay.ts

```diff
@@ -161,15 +196,45 @@ export class RelayImpl implements Relay {
       this.cacheService,
     );
 
+    this.hbarSpendingPlanRepository = new HbarSpendingPlanRepository(
+      this.cacheService,
+      logger.child({ name: 'hbar-spending-plan-repository' }),
+    );
+
+    this.ethAddressHbarSpendingPlanRepository = new EthAddressHbarSpendingPlanRepository(
+      this.cacheService,
+      logger.child({ name: 'eth-address-hbar-spending-plan-repository' }),
+    );
+
+    this.ipAddressHbarSpendingPlanRepository = new IPAddressHbarSpendingPlanRepository(
+      this.cacheService,
+      logger.child({ name: 'ip-address-hbar-spending-plan-repository' }),
+    );
```

Comment:
Since these repository instances are only used within the constructor and not elsewhere in the code, I believe initializing them as local constants is sufficient, rather than declaring them as class properties. This reduces their scope and save some memory, as they are not needed outside the constructor.

```suggestion
    const hbarSpendingPlanRepository = new HbarSpendingPlanRepository(
      this.cacheService,
      logger.child({ name: 'hbar-spending-plan-repository' }),
    );
    const ethAddressHbarSpendingPlanRepository = new EthAddressHbarSpendingPlanRepository(
      this.cacheService,
      logger.child({ name: 'eth-address-hbar-spending-plan-repository' }),
    );
    const ipAddressHbarSpendingPlanRepository = new IPAddressHbarSpendingPlanRepository(
      this.cacheService,
      logger.child({ name: 'ip-address-hbar-spending-plan-repository' }),
    );
```